src/nvim/mouse.c

```diff
@@ -487,25 +487,22 @@ bool do_mouse(oparg_T *oap, int c, int dir, int count, bool fixindent)
       if (is_click && cmdwin_type == 0 && mouse_col < Columns) {
         in_tab_line = true;
         c1 = tab_page_click_defs[mouse_col].tabnr;
+
+        // Check for modifiers when clicking the tab line.
+        // Middle mouse button in the tab line: Close the tab page.
+        if (which_button == MOUSE_MIDDLE
+            && !(mod_mask &
+                 (MOD_MASK_MULTI_CLICK | MOD_MASK_SHIFT | MOD_MASK_CTRL | MOD_MASK_ALT |
+                  MOD_MASK_META))) {
+          do_tabclose(c1);
+          return false;
```

Comment:
```suggestion
          return true;
```

Nit: doesn't matter here since middle mouse in insert mode (where `start_arrow()` applies) puts instead, but maybe it's best to stay consistent with the other code that returns `true` below?